### PR TITLE
fix: Fix a batch of 11 bugs from the active Athena reports

### PR DESCRIPTION
**Pull request by kristofbolyai**
I looked through around 300 reports, these are the relevant ones. I believe this should make the mod close to "perfect", in terms of crashes, at the moment.

### CODE REVIEW

File: common/src/main/java/com/wynntils/models/marker/UserWaypointMarkerProvider.java [hunk 1]
```diff
@@ -15,12 +15,12 @@
...
 import com.wynntils.utils.render.Texture;
 import com.wynntils.utils.type.Pair;
-import java.util.LinkedHashSet;
 import java.util.Set;
+import java.util.concurrent.CopyOnWriteArraySet;
 import java.util.stream.Stream;
 
 public class UserWaypointMarkerProvider implements MarkerProvider<WaypointPoi> {
-    private final Set<Pair<MarkerInfo, WaypointPoi>> markerInfoSet = new LinkedHashSet<>();
+    private final Set<Pair<MarkerInfo, WaypointPoi>> markerInfoSet = new CopyOnWriteArraySet<>();
```
Comment (magicus) [inline common/src/main/java/com/wynntils/models/marker/UserWaypointMarkerProvider.java:23]: Can you expand a bit about why this was needed? Preferably as a short comment in the code.

**Review by magicus — APPROVED** [review]
Great work! All fixes look good as far as I can tell.

File: common/src/main/java/com/wynntils/models/marker/UserWaypointMarkerProvider.java (hunk 1 above)
Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/models/marker/UserWaypointMarkerProvider.java:23]: Well, the marker provider implementations need to be race-proof.

Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/models/marker/UserWaypointMarkerProvider.java:23]: https://github.com/Wynntils/Artemis/pull/2125/commits/0488ef8de25077db8aa29a79b49be1990d715caf

**Pull request merged**